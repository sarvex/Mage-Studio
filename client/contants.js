export const ELEMENTS = {
    BASE: {
        CUBE: "CUBE",
        SPHERE: "SPHERE",
        CYLINDER: "CYLINDER",
        CONE: "CONE",
        BOX: "BOX",
        PLANE: "PLANE",
    },
    LIGHTS: {
        AMBIENT: "AMBIENT",
        SUN: "SUN",
    },
};

export const ELEMENTS_LABELS = {
    BASE: {
        CUBE: "Cube",
        SPHERE: "Sphere",
        CYLINDER: "Cylinder",
        CONE: "Cone",
        BOX: "Box",
        PLANE: "Plane",
    },
    LIGHTS: {
        AMBIENT: "Ambient",
        SUN: "Sun ",
    },
};
