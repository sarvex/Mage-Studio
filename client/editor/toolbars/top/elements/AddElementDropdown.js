import React from "react";
import { Dropdown, Button, Menu } from "antd";
import { PlusOutlined } from "@ant-design/icons";

import style from "../toolbar.module.scss";
import { ELEMENTS, ELEMENTS_LABELS } from "../../../../contants";
const { Item, SubMenu } = Menu;

export const getDropdownOverlay = onItemSelect => (
    <Menu onClick={onItemSelect}>
        <SubMenu title="Model">
            <Item>Import</Item>
        </SubMenu>
        <SubMenu title="Element">
            {Object.keys(ELEMENTS.BASE).map(k => (
                <Item key={k} title={k}>
                    {ELEMENTS_LABELS.BASE[k]}
                </Item>
            ))}
        </SubMenu>
        <SubMenu title="Sound">
            <Item>sorry</Item>
        </SubMenu>
        <SubMenu title="Light">
            {Object.keys(ELEMENTS.LIGHTS).map(k => (
                <Item key={k} title={k}>
                    {ELEMENTS_LABELS.LIGHTS[k]}
                </Item>
            ))}
        </SubMenu>
    </Menu>
);

const AddElementDropdown = ({ onElementSelection = f => f }) => {
    const onItemSelect = ({ key }) => {
        onElementSelection(key);
    };

    return (
        <div className={style["controls-settings-group"]}>
            <Dropdown
                overlay={getDropdownOverlay(onItemSelect)}
                trigger={["click"]}
                placement={"topLeft"}
            >
                <Button className={style["add-element-button"]}>
                    <PlusOutlined />
                    Add
                </Button>
            </Dropdown>
        </div>
    );
};

export default AddElementDropdown;
