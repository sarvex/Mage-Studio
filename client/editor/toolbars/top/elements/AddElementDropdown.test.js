import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

import AddElementDropdown, { AddMenu } from "./AddElementDropdown";
import { ELEMENTS } from "../../../../contants";
import sinon from "sinon";
import { Menu } from "antd";

describe("AddElementDropdown", () => {
    describe("render", () => {
        it("should render properly", () => {
            const component = shallow(<AddElementDropdown />);
            expect(toJSON(component)).toMatchSnapshot();
        });
    });

    // describe("AddMenu", () => {
    //     it("click on add -> mesh -> cube", () => {
    //         const spy = sinon.spy();
    //         const component = shallow(<AddMenu onItemSelect={spy} />);

    //         // console.log(component.debug());

    //         const item = component.findWhere(n => n.prop("title") === ELEMENTS.BASE.CUBE);
    //         item.simulate("click");

    //         console.log(spy.called);
    //         expect(spy.calledWith({ key: ELEMENTS.BASE.CUBE })).toBe(true);
    //     });
    // });
});
