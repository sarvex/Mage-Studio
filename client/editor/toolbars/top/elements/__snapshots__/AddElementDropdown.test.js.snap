// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddElementDropdown render should render properly 1`] = `
<div
  className="controls-settings-group"
>
  <Dropdown
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    overlay={
      <Menu
        onClick={[Function]}
      >
        <SubMenu
          title="Model"
        >
          <MenuItem>
            Import
          </MenuItem>
        </SubMenu>
        <SubMenu
          title="Element"
        >
          <MenuItem
            title="CUBE"
          >
            Cube
          </MenuItem>
          <MenuItem
            title="SPHERE"
          >
            Sphere
          </MenuItem>
          <MenuItem
            title="CYLINDER"
          >
            Cylinder
          </MenuItem>
          <MenuItem
            title="CONE"
          >
            Cone
          </MenuItem>
          <MenuItem
            title="BOX"
          >
            Box
          </MenuItem>
          <MenuItem
            title="PLANE"
          >
            Plane
          </MenuItem>
        </SubMenu>
        <SubMenu
          title="Sound"
        >
          <MenuItem>
            sorry
          </MenuItem>
        </SubMenu>
        <SubMenu
          title="Light"
        >
          <MenuItem
            title="AMBIENT"
          >
            Ambient
          </MenuItem>
          <MenuItem
            title="SUN"
          >
            Sun 
          </MenuItem>
        </SubMenu>
      </Menu>
    }
    placement="topLeft"
    trigger={
      Array [
        "click",
      ]
    }
  >
    <Button
      className="add-element-button"
    >
      <ForwardRef(PlusOutlined) />
      Add
    </Button>
  </Dropdown>
</div>
`;
